app/src/main/java/org/oppia/app/home/continueplaying/ContinuePlayingFragmentPresenter.kt

```diff
@@ -27,71 +26,61 @@ class ContinuePlayingFragmentPresenter @Inject constructor(
+      it.viewModel = viewModel
     }
-
-    subscribeToOngoingStoryList()
-
     return binding.root
   }
 
-  private val ongoingStoryListSummaryResultLiveData: LiveData<AsyncResult<OngoingStoryList>> by lazy {
-    topicListController.getOngoingStoryList()
-  }
-
-  private fun subscribeToOngoingStoryList() {
-    getAssumedSuccessfulOngoingStoryList().observe(fragment, Observer<OngoingStoryList> { it ->
-      if (it.recentStoryCount > 0) {
-        val recentSectionTitleViewModel =
-          SectionTitleViewModel(activity.getString(R.string.ongoing_story_last_week), false)
-        itemList.add(recentSectionTitleViewModel)
-        for (promotedStory in it.recentStoryList) {
-          val ongoingStoryViewModel = OngoingStoryViewModel(promotedStory, fragment as OngoingStoryClickListener)
-          itemList.add(ongoingStoryViewModel)
+  private fun createRecyclerViewAdapter(): BindableAdapter<ContinuePlayingItemViewModel> {
+    return BindableAdapter.MultiTypeBuilder
+      .newBuilder<ContinuePlayingItemViewModel, ViewType> { viewModel ->
+        when (viewModel) {
+          is SectionTitleViewModel -> ViewType.VIEW_TYPE_SECTION_TITLE_TEXT
+          is OngoingStoryViewModel -> ViewType.VIEW_TYPE_SECTION_STORY_ITEM
+          else -> throw IllegalArgumentException("Encountered unexpected view model: $viewModel")
         }
       }
+      .registerViewDataBinder(
+        viewType = ViewType.VIEW_TYPE_SECTION_TITLE_TEXT,
+        inflateDataBinding = SectionTitleBinding::inflate,
+        setViewModel = SectionTitleBinding::setViewModel,
+        transformViewModel = { it as SectionTitleViewModel }
+      )
+      .registerViewDataBinder(
+        viewType = ViewType.VIEW_TYPE_SECTION_STORY_ITEM,
+        inflateDataBinding = OngoingStoryCardBinding::inflate,
+        setViewModel = OngoingStoryCardBinding::setViewModel,
+        transformViewModel = { it as OngoingStoryViewModel }
+      )
+      .build()
+  }
 
-      if (it.olderStoryCount > 0) {
-        val showDivider = itemList.isNotEmpty()
-        val olderSectionTitleViewModel =
-          SectionTitleViewModel(activity.getString(R.string.ongoing_story_last_month), showDivider)
-        itemList.add(olderSectionTitleViewModel)
-        for (promotedStory in it.olderStoryList) {
-          val ongoingStoryViewModel = OngoingStoryViewModel(promotedStory, fragment as OngoingStoryClickListener)
-          itemList.add(ongoingStoryViewModel)
-        }
-      }
-      ongoingListAdapter.notifyDataSetChanged()
-    })
+  private fun getContinuePlayModel(): ContinuePlayViewModel {
+    return viewModelProvider.getForFragment(fragment, ContinuePlayViewModel::class.java)
   }
 
-  private fun getAssumedSuccessfulOngoingStoryList(): LiveData<OngoingStoryList> {
-    // If there's an error loading the data, assume the default.
-    return Transformations.map(ongoingStoryListSummaryResultLiveData) { it.getOrDefault(OngoingStoryList.getDefaultInstance()) }
+  private enum class ViewType {
```

Comment:
Shouldn't this be in a separate file?

Reply:
I can put it inside ContinuePlayingItemViewModel.

Reply:
Done.